Yosemite/Yosemite/Stores/Order/OrdersUpsertUseCase.swift

```diff
@@ -258,4 +259,27 @@ struct OrdersUpsertUseCase {
             }
         }
     }
+
+    /// Updates, inserts, or prunes the provided `storageOrder`'s taxes using the provided `readOnlyOrder`'s taxes
+    ///
+    private func handleOrderTaxes(_ readOnlyOrder: Networking.Order, _ storageOrder: Storage.Order, _ storage: StorageType) {
+        // Upsert the `taxes` from the `readOnlyOrder`
+        readOnlyOrder.taxes.forEach { readOnlyTax in
+            if let existingStorageTax = storage.loadOrderTaxLine(siteID: readOnlyOrder.siteID, taxID: readOnlyTax.taxID) {
+                existingStorageTax.update(with: readOnlyTax)
+            } else {
+                let newStorageTax = storage.insertNewObject(ofType: Storage.OrderTaxLine.self)
+                newStorageTax.update(with: readOnlyTax)
+                storageOrder.addToTaxes(newStorageTax)
+            }
+        }
+
+        // Now, remove any objects that exist in `storageOrder.taxes` but not in `readOnlyOrder.taxes`
+        storageOrder.taxes?.forEach { storageFee in
+            if readOnlyOrder.taxes.first(where: { $0.taxID == storageFee.taxID } ) == nil {
+                storageOrder.removeFromTaxes(storageFee)
+                storage.deleteObject(storageFee)
+            }
```

Comment:
I think `storageFee` should be replaced with `storageTax` here 😄

Reply:
🤦 Fixed in - https://github.com/woocommerce/woocommerce-ios/pull/5931/commits/a8cd4ad6793c18bc7b375beb0c56905624254ed5